### ISSUE TITLE
pokemon-showdown simulate-battle broken on mac and linux

### ISSUE THREAD

**Issue by curtdissel** [smogon/pokemon-showdown#7600]
Not sure which PR triggered it. But simulate-battle no longer works when i try it on linux and mac.

I never had issues with the Node10 version of showdown and simulate battle. Only after switching to the Node12 version.
I think the showdown server itself works fine. Just simulate-battle is broken on Mac and Linux. Didn't try windows yet.

If i use node10 + this commit, it works fine.
git checkout 529fdaaa685d8aa695fba01a80cccb3c1d761f7c

Commands that i run that generate failure:
git clone https://github.com/Zarel/Pokemon-Showdown
./Pokemon-Showdown/pokemon-showdown simulate-battle

Error:
3 packages are looking for funding
  run `npm fund` for details

found 4 vulnerabilities (3 low, 1 high)
  run `npm audit fix` to fix them, or `npm audit` for details
config.js does not exist. Creating one with default settings...
internal/modules/cjs/loader.js:834
  throw err;
  ^

Error: Cannot find module './dex'
Require stack:
- /Pokemon-Showdown/.sim-dist/battle.js
- /Pokemon-Showdown/.sim-dist/battle-stream.js
- /Pokemon-Showdown/pokemon-showdown
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:831:15)
    at Function.Module._load (internal/modules/cjs/loader.js:687:27)
    at Module.require (internal/modules/cjs/loader.js:903:19)
    at require (internal/modules/cjs/helpers.js:74:18)
    at Object.<anonymous> (/Pokemon-Showdown/.sim-dist/battle.js:7:12)
    at Module._compile (internal/modules/cjs/loader.js:1015:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1035:10)
    at Module.load (internal/modules/cjs/loader.js:879:32)
    at Function.Module._load (internal/modules/cjs/loader.js:724:14)
    at Module.require (internal/modules/cjs/loader.js:903:19) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/Pokemon-Showdown/.sim-dist/battle.js',
    '/Pokemon-Showdown/.sim-dist/battle-stream.js',
    '/Pokemon-Showdown/pokemon-showdown'
  ]
}

**Comment by AnnikaCodes** [comment]
I use macOS and node 14, and it works fine for me. Have you run `node build`?

**Comment by curtdissel** [comment]
Interesting. I can confirm that node 14 does appear to work on my Mac as well. Thanks for the helpful tip. On Ubuntu, I still get the same error.

I will try to play around with installs on the ubuntu to see if i come across a version that works.

**Comment by curtdissel** [comment]
Mac is good with Node 14LTS.

Ubuntu18 I tried: Node 14/15/LTS
No good with simulate battle.

I may try another distro tomorrow.

**Comment by Zarel** [comment]
`./pokemon-showdown simulate-battle`, for performance reasons, does not automatically rebuild. You do need to run `./build` after updating, unfortunately.

**Comment by Zarel** [comment]
Your error message tells me you need to run `./build --force`. In theory you should never need to force-build, but we seem to have some bugs that require it.

**Comment by curtdissel** [comment]
Thanks so much you two. the build --force was key. It seems AnnikaCodes did mention node build, but i interpreted what she said incorrectly. I typed npm build or something like that. Had I typed node build like she said, it probably would have been resolved sooner.

As a token of my gratitude to the two of you. Soon I will be looking for participants for Pokemon Ai. I will keep two slots open for you two. My Pokemon Sword tech is almost stable enough for private beta.

I would strongly recommend you start becoming familiar with Pytorch or Tensorflow.

Long story short, I have bots that can play sword shield matches on nintendo switch. I also collect replay data from live matches to train neural networks etc. if you want to verify my abilities, check out some of my past broadcast on twitch. PokeRandoBoto. I have other bots as well. The video quality is poor, but observe the chat. The chat is proof of the bot.
